File: services/reservation/src/main/java/com/example/iter/reservation/dto/response/RentalApproveResponse.java
package com.example.iter.reservation.dto.response;

import com.example.iter.reservation.domain.entity.Rental;
import com.example.iter.reservation.api.RentalStatus;

import java.time.LocalDateTime;

public record RentalApproveResponse(
        Long rentalId,
        RentalStatus status,
        LocalDateTime approvedAt
) {
    public static RentalApproveResponse from(Rental rental) {
        return new RentalApproveResponse(rental.getId(), rental.getStatus(), rental.getApprovedAt());
    }
}

File: services/reservation/src/main/kotlin/com/example/iter/reservation/dto/response/RentalApproveResponse.kt
package com.example.iter.reservation.dto.response

import com.example.iter.reservation.api.RentalStatus
import com.example.iter.reservation.domain.entity.Rental

import java.time.LocalDateTime

data class RentalApproveResponse(
    val rentalId: Long?,
    val status: RentalStatus?,
    val approvedAt: LocalDateTime?,
) {
    fun rentalId(): Long? = rentalId
    fun status(): RentalStatus? = status
    fun approvedAt(): LocalDateTime? = approvedAt

    companion object {
        @JvmStatic
        fun from(rental: Rental): RentalApproveResponse =
            RentalApproveResponse(rental.id, rental.status, rental.approvedAt)
    }
}
